### ISSUE TITLE
vim.lsp.util._str_utfindex_enc wrapper has been removed from neovim, but is still used in this repo

### ISSUE THREAD

**Issue by magnusriga** [pmizio/typescript-tools.nvim#311]
Thanks for a solid tool. I have come across the below error.

The `vim.lsp.util._str_utfindex_enc` wrapper function has been removed from nvim (https://github.com/neovim/neovim/pull/30915), but is still used here:

https://github.com/pmizio/typescript-tools.nvim/blob/1edc5dcba8cb46e71feddde91c9c3d4bce77bec5/lua/typescript-tools/protocol/text_document/semantic_tokens.lua#L30

Any chance for a quick fix?

The function just needs to be switched with the new `vim.str_utfindex(s, encoding, index, strict_indexing)`.

In other words, I believe the call above to `vim.lsp.util._str_utfindex_enc` can be swapped out with:

`vim.str_utfindex(line, "utf-16", nil, false)`.

**Comment by KostkaBrukowa** [comment]
Added a PR with a fix, but I will use this fix for a day on nightly build to check if everything works as expected
